ui/src/app/views/workspace-edit.tsx

```diff
@@ -390,8 +434,45 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
         cloneUserRole: false,
         loading: false,
         showUnderservedPopulationDetails: false,
-        showStigmatizationDetails: false
+        showStigmatizationDetails: false,
+        autocompleteLoading: false,
+        autocompleteDiseases: [],
+        diseaseHover: [],
+        dropDown: false,
+        searchTerm: '',
+        hovering: false
       };
+      this.searchTermChangedEvent = fp.debounce(300, this.diseaseSearch);
+    }
+
+    get showSearchResults(): boolean {
+      return !this.state.autocompleteLoading &&
+        this.state.autocompleteDiseases.length > 0 &&
+        !isBlank(this.state.searchTerm) && this.state.dropDown;
+    }
+
+    diseaseSearch(value: string): void {
+      this.setState({autocompleteLoading: true, autocompleteDiseases: [], searchTerm: value});
+      if (!value.trim()) {
+        this.setState({autocompleteLoading: false, dropDown: false});
+        return;
+      }
+      const self = this;
+      const url = 'https://firecloud-orchestration.dsde-dev.broadinstitute.org' +
+        '/duos/autocomplete/' + value;
+      fetch(encodeURI(url)).then((response) => {
+        return response.json();
+      }).then((matches) => {
+        const labeledMatches = fp.filter((elt) => elt.hasOwnProperty('label'))(matches);
+        const labels = fp.map((elt) => elt['label'])(labeledMatches);
+        const diseaseHover = labels.map(() => false);
+        self.setState({
```

Comment:
Curious why you need to bind self above? Because you're just using lambdas, the `this` should be preserved in here.

Reply:
Not needed anymore, will use `this`